iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheck.java

```diff
@@ -156,6 +157,12 @@ private static BiConsumer<CheckContext, ResourceDeclaration> checkPublicNetworkA
       .ifPresent(value -> ctx.reportIssue(value, PUBLIC_NETWORK_ACCESS_MESSAGE));
   }
 
+  private static BiConsumer<CheckContext, ResourceDeclaration> checkPublicNetworkAccessInNetwork() {
+    return (ctx, resource) -> CheckUtils.resolveProperties("network/publicNetworkAccess", resource).stream()
+      .filter(PublicNetworkAccessCheck::isSensitivePublicNetworkAccessSimplified)
+      .forEach(value -> ctx.reportIssue(value, PUBLIC_NETWORK_ACCESS_MESSAGE));
+  }
```

Comment:
```suggestion
  private static void checkPublicNetworkAccessInNetwork(ContextualResource resource) {
    resource
      .object("network")
      .property("publicNetworkAccess")
      .reportIf(PublicNetworkAccessCheck::isSensitivePublicNetworkAccessSimplified, PUBLIC_NETWORK_ACCESS_MESSAGE);
  }
```
Suggestion with the new DSL, I find it more easy to read.
No need to change it now, I let you decide if you want to do it, and in that case it would probably make more sense to do it in another ticket for the whole rule.